File: Niveau 3/Niveau 3.8/retournement.hs
--------------------------------------------------------------------------------
-- Name:        Retournement de chaîne
-- Purpose:     France-IOI niveau 3.8
--
-- Author:      Sébastien
--
-- Created:     04/10/2022
--------------------------------------------------------------------------------

main :: IO()
main = do  
    chaine <- getLine
    putStrLn $ retourner' chaine
    
retourner :: String -> String
retourner [] = []
retourner (x:xs) = (retourner xs)++[x]

retourner' :: String -> String
retourner' chaine = foldl (\x y -> y:x) [] chaine

File: Niveau 3/Niveau 3.8/retournement.py
#-------------------------------------------------------------------------------
# Name:        Retournement de chaîne
# Purpose:     France-IOI niveau 3.8
#
# Author:      Sébastien
#
# Created:     04/10/2022
#-------------------------------------------------------------------------------

def retourner(chaine):
    if len(chaine) > 0:
        retourner(chaine[1:])
        print(chaine[0], end='')

if __name__ == '__main__':
    chaine = input()
    retourner(chaine)
    print()
